File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int x = INT_MAX;

        for (int r = 0; r < prices.size(); ++r) {
            x = min(x, prices[r]);
            ans = max(ans, prices[r] - x);
        }

        return ans;
    }
};

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total = 0
        cur = prices[0]

        for i in range(1, len(prices)):
            if cur < prices[i]:
                total += prices[i] - cur

            cur = prices[i]

        return total